ls1intum/Artemis
# `Development`: Add API versioning

### Checklist
#### General


#### Server

### Motivation and Context


For future apps and other external applications, we want to make the API a bit more resilient to breaking changes. Hence, we want to introduce a versioned API.

### Description

I introduced a request mapper that gets injected into the auto configuration. This mapper resolves all endpoints according to their optional versioning annotation. I added extensive JavaDoc and a small documentation. The API prefix `/api` is now part of the request mapper and hence all such annotations were removed from the controllers.
I also added a test that validates the setup of all endpoints in regards to versioning.

Currently, we deactivated the rendering of the API documentation due to performance issues. In the future we want to render the documentation in the CI and publish it similarly to our other documentation: #5932

### Steps for Testing

Changes cover all of the system but not in a functional way. It is expected to be fully backwards-compatible so please try out some stuff you would usually do and check that there are no issues. There are no in-detail tests necessary.


#### Exam Mode Testing


Changes cover all of the system but not in a functional way. It is expected to be fully backwards-compatible so please try out some stuff you would usually do and check that there are no issues. There are no in-detail tests necessary.


### Review Progress






#### Code Review
- [x] Review 4
#### Manual Tests
